CONTRIBUTING.md

```diff
@@ -0,0 +1,124 @@
+- Make use of macros (ours and dbt's) where possible to avoid duplication and to manage the differences between warehouses
+  - Do not reinvent the wheel e.g. make use of [`type_*` macros](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros#data-type-functions) instead of explicit datatypes
+  - In the case where a macro may be useful outside of a specific package, we may make the choice to add it to `dbt-snowplow-utils` instead
+- Make use of the incremental logic as much as possible to avoid full-scanning large tables
+- Where new functionality is being added, or you are touching existing functionality that does not have good/any test, add tests
+
+## Testing
+
+Once you're able to manually test that your code change is working as expected, it's important to run existing automated tests, as well as adding some new ones. These tests will ensure that:
+- Your code changes do not unexpectedly break other established functionality
+- Your code changes can handle all known edge cases
+- The functionality you're adding will _keep_ working in the future
+
+In general our packages all have similar structures, with an `integration_tests` folder that contains a `.scripts/integration_tests.sh` file. This script is run with 1 argument, the name of your `target` in the `integration_tests` profile e.g. `./integration_tests/.scripts/integration_tests.sh -d postgres` which will run all the tests on your postgres instance. This all means you don't need your own Snowplow data to run the tests.
+
+Tests are of 1 of 2 kinds:
+- Row count/equality tests; these ensure that the processed seed data from the package matches exactly an expected input seed file. If you have made no change to logic these should not fail, however if you have changed the logic you may need to edit the expected seed file, and add records to the events input seed file to cover the use case. In some cases it may make sense to add both expected and unexpected data to the test (i.e. to ensure a fix you have deployed actually fixes the issue you have seen).
+- Macro based tests; these are more varied, sometimes checking the output sql from a macro or otherwise examining database objects. Look at existing tests for more details and for how to edit/create these.
+
+To run the integration tests:
+1. Ensure the `integration_tests` folder is your working directory (you may need to `cd integration_tests`)
+2. Run `dbt run-operation post_ci_cleanup` to ensure a clean set of schemas (this will drop the schemas we use, so ensure your profile is only for these tests)
+3. Run `./.scripts/integration_test.sh -d {target}` with your target name
+4. Ensure all tests run successfully
+
+If any tests fail, you should examine the outputs and either correct the test or correct your changes.
+
+> If you do not have access to all warehouses do not worry, test what you can and the remainder will be run when you submit your PR.
```

Comment:
Not technically true since the GH actions won't run when a member outside of our org makes a PR afaik, but not sure if worth clarifying